remote_attestation/rust/src/message.rs

```diff
@@ -235,3 +270,62 @@ impl Deserializable for ClientIdentity {
         }
     }
 }
+
+impl EncryptedData {
+    pub fn new(nonce: &[u8; NONCE_LENGTH], data: &[u8]) -> Self {
+        Self {
+            header: ENCRYPTED_DATA_HEADER,
+            nonce: *nonce,
+            data: data.to_vec(),
+        }
+    }
+}
+
+impl Serializable for EncryptedData {
+    fn serialize(&self) -> anyhow::Result<Vec<u8>> {
+        bincode::serialize(&self).context("Couldn't serialize encrypted data message")
```

Comment:
I am a bit concerned about the blackbox nature of bincode. How does it handle unexpected data such as incorrect message lengths? Does it truncate/pad the resulting arrays, or return an error? Will this behaviour stay the same in future? I think it is worth considering manual serialisation and deserialisation similar to the java implementation.

Reply:
+1